.github/workflows/test-and-push-docker-image.yaml

```
@@ -0,0 +1,122 @@
+name: Run tests and push Docker image on success
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+
+  test-js:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [8.x, 10.x, 12.x]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+
+      - name: Install JavaScript dependencies
+        run: |
+          npm install
```

```suggestion
        run: npm install
```

This can just be one line to tidy things up slightly. Maybe?

---

🤔 😬  I kinda liked the consistency of these but I'll move to same line when there is only 1 command if it pleases you ☺️